src/reader.rs

```diff
@@ -50,6 +48,15 @@ fn first_char(input: &str) -> char {
     input.chars().next().unwrap()
 }
 
+/// Returns a Some("-"), when available, or 'None'
+fn maybe_minus(input: &str) -> Option<&str>{
```

Comment:
Last change;  I assume this function was part of another attempt?  This can be safely removed I assume, everything else looks great!

Reply:
good point! removed and squished